### DISCUSSION TITLE
Does this library has iProxy implementation ?

### DISCUSSION THREAD

**Discussion by mega6453** [artehe/Netimobiledevice#32] [Q&A]
Can we proxy local port to iOS port using this library?
E.g. iproxy 5555 9100
Here 5555 is local windows port, 9100 is iOS port.

**Comment by artehe** [comment]
Sort of.

You can't specify the windows port specifically, but you can connect to an open iOS port. A rough example I've used before is this:

```csharp
public static async Task ConnectAndMessageTest(UsbmuxdDevice device)
{
    Socket conn = device.Connect(9100);

    for (byte i = 0; i < 10; i++) {
        byte[] buffer = new byte[1] { i };
        int dataSent = await conn.SendAsync(buffer, SocketFlags.None);
        byte[] recBuffer = new byte[dataSent];
        int response = await conn.ReceiveAsync(recBuffer, SocketFlags.None);
        if (response != dataSent) {
            throw new DataMisalignedException();
        }
    }

    conn.Close();
}
```

**Comment by mega6453** [comment]
oh ok. I'm expecting something like iproxy where I can specify which windows local port it should use. But I'm fine. Thanks for the reply.
